File: ios/LocationSharingApp/LocationSharingApp/Models/Friend.swift
import Foundation

struct Friend: Codable, Identifiable {
    let id: String
    let user: User
    let friendshipStatus: FriendshipStatus
    let locationSharingEnabled: Bool
    let lastKnownLocation: Location?
    let locationSharingPermission: LocationSharingPermission
    
    init(id: String, user: User, friendshipStatus: FriendshipStatus, locationSharingEnabled: Bool, lastKnownLocation: Location? = nil, locationSharingPermission: LocationSharingPermission) {
        self.id = id
        self.user = user
        self.friendshipStatus = friendshipStatus
        self.locationSharingEnabled = locationSharingEnabled
        self.lastKnownLocation = lastKnownLocation
        self.locationSharingPermission = locationSharingPermission
    }
}

enum FriendshipStatus: String, Codable, CaseIterable {
    case pending = "PENDING"
    case accepted = "ACCEPTED"
    case blocked = "BLOCKED"
}

enum LocationSharingPermission: String, Codable, CaseIterable {
    case none = "NONE"
    case requested = "REQUESTED"
    case granted = "GRANTED"
    case denied = "DENIED"
}

File: android/app/src/main/java/com/locationsharing/app/data/models/Friend.kt
package com.locationsharing.app.data.models

data class Friend(
    val id: String,
    val user: User,
    val friendshipStatus: FriendshipStatus,
    val locationSharingEnabled: Boolean,
    val lastKnownLocation: Location? = null,
    val locationSharingPermission: LocationSharingPermission
)

enum class FriendshipStatus {
    PENDING, ACCEPTED, BLOCKED
}

enum class LocationSharingPermission {
    NONE, REQUESTED, GRANTED, DENIED
}